core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/devui/ChatJsonRPCService.java

```diff
@@ -62,14 +65,17 @@ public class ChatJsonRPCService {
 
     private final List<ToolSpecification> toolSpecifications;
     private final Map<String, ToolExecutor> toolExecutors;
+    private final ToolProvider toolProvider;
 
     public ChatJsonRPCService(@All List<ChatLanguageModel> models, // don't use ChatLanguageModel model because it results in the default model not being configured
             @All List<StreamingChatLanguageModel> streamingModels,
             @All List<Supplier<RetrievalAugmentor>> retrievalAugmentorSuppliers,
             @All List<RetrievalAugmentor> retrievalAugmentors,
             ChatMemoryProvider memoryProvider,
-            QuarkusToolExecutorFactory toolExecutorFactory) {
+            QuarkusToolExecutorFactory toolExecutorFactory,
+            @All List<ToolProvider> toolProviders) {
```

Comment:
Maybe you should request the list of all `Supplier<ToolProvider>`s instead?
IIUC a `ToolProvider` isn't required to be a CDI bean, so if it's not, then it won't appear here. But the suppliers are required to be CDI beans.